### PR TITLE
Fix Windows terminal handling

### PR DESCRIPTION
**Pull request by ncdc**
Fix some issues with Windows terminal handling with respect to TTYs that came up as part of the
code that adds support for terminal resizing.

cc @smarterclayton @sttts @csrwng

### CODE REVIEW

File: pkg/kubectl/cmd/exec.go [hunk 1]
```diff
@@ -177,6 +183,47 @@ func (p *ExecOptions) Validate() error {
...
+
+		return t
+	}
+
+	// if we get to here, the user wants to attach stdin, wants a TTY, and o.In is a terminal, so we
+	// can safely set t.Raw to true
+	t.Raw = true
+
+	// use dockerterm.StdStreams() to get the right I/O handles on Windows
+	stdin, stdout, _ := dockerterm.StdStreams()
```
Comment (smarterclayton) [inline pkg/kubectl/cmd/exec.go]: This needs to be exposed so it is testable.  Would be sufficient to have a member function "func overrideStreams() (io.Reader, io.Writer, io.Writer)" that can be substituted.

**Comment by smarterclayton** [comment]
One comment, everything else looks ok.

File: pkg/kubectl/cmd/exec.go [hunk 2]
```diff
@@ -195,26 +242,10 @@ func (p *ExecOptions) Run() error {
...
-	if p.Stdin {
-		t.In = p.In
-		if tty && !t.IsTerminalIn() {
-			tty = false
-			fmt.Fprintln(p.Err, "Unable to use a TTY - input is not a terminal or the right kind of file")
-		}
-	} else {
-		p.In = nil
-	}
-	t.Raw = tty
```
Comment (sttts) [inline pkg/kubectl/cmd/exec.go]: Note that the logic changed here for `!p.Stdin && p.TTY`. I guess that's intentional, i.e. that `!p.Stdin => !t.Raw` holds now.

Wondering about having an output-only container without the need to have a stdin...

**Comment by sttts** [comment]
Same for me, one comment/question. Otherwise, lgtm.

File: pkg/kubectl/cmd/exec.go (hunk 2 above)
Comment (ncdc) [inline pkg/kubectl/cmd/exec.go]: I've thought about no stdin, tty=true, output only. I think the use cases for that combination are slim to none. You ok with this change as-is?

Comment (sttts) [inline pkg/kubectl/cmd/exec.go]: Yes, I agree about the use cases.

**Comment by ncdc** [comment]
@smarterclayton @sttts updated, PTAL

**Comment by ncdc** [comment]
@k8s-bot e2e test this issue: #29308

**Pull request merged**